### ISSUE TITLE
[table] focus indicator can get out of sync when using an NVDA screenreader with arrow keys on an interactive table

### ISSUE THREAD

**Issue by HelainaCurtis** [tyler-technologies-oss/forge#1111]
### Package

forge

### Version

3.0.0

### Description

We have a table pictured below that is highly interactive. I was able to get it to have the focus indicator on one checkbox of a row, and the screenreader was on a different row. When I hit space at that point, it checks the checkbox of the screenreader spot. I noticed that the table behaves different with the screenreader on than when it's off. 

<img width="1207" height="428" alt="Image" src="https://github.com/user-attachments/assets/1c719969-b4f2-44f1-9298-9a55f10f931d" />

### Steps to Reproduce

using storybook table demo as example
1. have NVDA screenreader running 
2.click the top right checkbox to select all to have focus in the table
3. tab 4 times to get to the first checkbox with Eve
4. down arrow key 4 times(screen reader should announce "row 3 first name Bob...")
5. hit space
Nothing happens.

### Expected Behavior

Either focus indicator should follow when using arrow keys to new interactive elements, or the arrow keys shouldn't work. I compared this to ag grid and in their case the arrow keys just didn't let me walk through the data.

### Environment

Edge
